pkg/trait/knative.go

```diff
@@ -560,3 +571,19 @@ func (t *knativeTrait) extractServices(names []string, serviceType knativeapi.Ca
 	sort.Strings(answer)
 	return answer
 }
+
+func (t *knativeTrait) determineServiceType(e *Environment) (string, error) {
+	controllerStrategy, err := e.DetermineControllerStrategy()
+	if err != nil {
+		return "", err
+	}
+
+	switch controllerStrategy {
+	case ControllerStrategyKnativeService:
+		return serving.SchemeGroupVersion.String(), nil
+	case ControllerStrategyDeployment:
```

Comment:
Probably also when CronJob strategy. In fact, why not defaulting to CreateTriggerService if not Knative enabled?

Reply:
I' d rather raise the error in this case because a CronJob deployment strategy does not fit the Trigger.